pdbeccdutils/tests/test_file_writing.py

```diff
@@ -209,3 +211,43 @@ def test_plain_cif_write(component: Component, tmpdir, rem_hs):
         assert component.id == cif_block.name
         for c in to_check:
             assert c in cif_block.get_mmcif_category_names()
+
+    @staticmethod
+    def test_fragment_writing(component: Component, tmpdir):
+        path = tmpdir.join(f"{component.id}.cif")
+        fragment_library = FragmentLibrary()
+        component.library_search(fragment_library)
+        ccd_writer.write_molecule(str(path), component)
+        cif_block = cif.read(str(path)).sole_block()
+        assert cif_block
+        assert component.id == cif_block.name
+        cif_categories = cif_block.get_mmcif_category_names()
+        if component.fragments:
+            assert "_pdbe_chem_comp_substructure." in cif_categories
+            substructure = cif_block.get_mmcif_category("_pdbe_chem_comp_substructure.")
+            for _, values in substructure.items():
+                for value in values:
+                    if value is not None:
+                        assert len(value) > 0
+
+    @staticmethod
+    def test_scaffold_writing(component: Component, tmpdir):
+        path = tmpdir.join(f"{component.id}.cif")
+        try:
+            component.get_scaffolds()
+            ccd_writer.write_molecule(str(path), component)
+            cif_block = cif.read(str(path)).sole_block()
+            assert cif_block
+            assert component.id == cif_block.name
+            cif_categories = cif_block.get_mmcif_category_names()
+            if component.scaffolds:
+                assert "_pdbe_chem_comp_substructure." in cif_categories
+                substructure = cif_block.get_mmcif_category(
+                    "_pdbe_chem_comp_substructure."
+                )
+                for _, values in substructure.items():
```

Comment:
Look at Oliver's test in the other pull request and may be think of how to incorporate it. It seems to test "''" which is a good test here.